src/net_processing.cpp

```diff
@@ -783,11 +799,8 @@ unsigned int LimitOrphanTxSize(unsigned int nMaxOrphans)
     while (mapOrphanTransactions.size() > nMaxOrphans)
     {
         // Evict a random orphan:
-        uint256 randomhash = rng.rand256();
-        std::map<uint256, COrphanTx>::iterator it = mapOrphanTransactions.lower_bound(randomhash);
-        if (it == mapOrphanTransactions.end())
-            it = mapOrphanTransactions.begin();
-        EraseOrphanTx(it->first);
+        size_t randompos = rng.randrange(g_orphan_list.size());
+        EraseOrphanTx(g_orphan_list[randompos]->first);
```

Comment:
@sipa 

As the `mapOrphanTransactions` items are sorted by hash, the first item is random with a negligible bias:

```suggestion
        EraseOrphanTx(mapOrphanTransactions.begin()->first);
```

Or did I miss something?

Reply:
That would always evict lower txids before higher txids. Not exactly negligible...

Reply:
Is this applicable
https://crypto.stackexchange.com/questions/12822/are-the-sha-family-hash-outputs-practically-random
?

Reply:
> That would always evict lower txids before higher txids. Not exactly negligible...

I do not mean "random txid", rather "random transaction".

Reply:
No. That would apply if the transactions were secret, but since an attacker may know the transactions, they also know the txids. The hashing applied is not relevant.

If `mapOrphanTransactions` was say an unordered_map, and was using salted hash for its *internal* hashing (to convert txids to bucket positions) like is used in some places, then this would perhaps not be an issue.